vortex-scalar/src/list.rs

```diff
@@ -104,21 +104,42 @@ impl<'a, T: for<'b> TryFrom<&'b Scalar, Error = VortexError>> TryFrom<&'a Scalar
     }
 }
 
-impl<T> From<Vec<T>> for Scalar
-where
-    Self: From<T>,
```

Comment:
I think you could keep it and instead do `ScalarValue: From<T>` then you'd avoid generating scalars since you only ever need values. What you have is also a bit different than the existing code. You could also put a bound on T: NativePType and you'd probabaly get the same result without having a macro

Reply:
So, this
```
impl<T: NativePType> From<Vec<T>> for Scalar
where ScalarValue: From<T>
```
compiles but it breaks vortex-array because usize is not NativePType.

The workspace compiles and cargo test passes if usize is a NativePType (where the PTYPE is U64). However match_each_native_ptype is now a bit weird because we lose the infectiveness of `NativePType<T>::PTYPE`.

Reply:
Happy to go that route if you prefer; I don't have enough context to know if the loss of injectiveness is bad